utilities/checkpoint/checkpoint_test.cc

```diff
@@ -985,6 +987,45 @@ TEST_F(CheckpointTest, CheckpointWithDbPath) {
   delete checkpoint;
 }
 
+TEST_F(CheckpointTest, CheckpointWithArchievedLog) {
+  std::atomic_bool flushed = false;
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->LoadDependency(
+      {{"WalManager::ArchiveWALFile",
+        "CheckpointTest:CheckpointWithArchievedLog"}});
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
+      "DBImpl::FlushForGetLiveFiles", [&](void*) { flushed = true; });
```

Comment:
The use of SyncPoint here is a little excessive, but OK. One downside of SyncPoint is that it's error-prone - the string name could have typo and that could make the test accidentally pass. Another downside is it's low-level - there could be a different code path flushing the memtable.

The more conventional alternative would be to check `NumTableFilesAtLevel(0)` is the same before and after the checkpoint. You could even make that a little bit more robust by setting `disable_auto_compactions=true` to prevent compaction out of L0.

Reply:
thx for advice, updated it